src/app/libp2p_helper/src/codanet.go

```diff
@@ -465,53 +465,49 @@ func (cv customValidator) Select(key string, values [][]byte) (int, error) {
 }
 
 func (h *Helper) handlePxStreams(s network.Stream) {
-	fromSeed := false
-
-	for _, seed := range h.Seeds {
-		if s.Conn().RemotePeer() == seed.ID {
-			fromSeed = true
-			break
-		}
+	connInfo := h.ConnectionManager.GetInfo()
+	if connInfo.ConnCount >= connInfo.LowWater {
+		_ = s.Close()
+		return
 	}
 
-	if !fromSeed {
-		logger.Debugf("ignoring peer-exchange stream from non-seed peer=%s", s.Conn().RemotePeer())
+	buf := make([]byte, 8192)
+	_, err := s.Read(buf)
+	if err != nil && err != io.EOF {
+		logger.Errorf("failed to decode list of peers err=%s", err)
 		_ = s.Close()
 		return
 	}
 
-	for {
-		dec := json.NewDecoder(s)
-		peers := []peer.AddrInfo{}
-		err := dec.Decode(&peers)
-		if err != nil && err == io.EOF {
-			continue
-		} else if err != nil && err.Error() == "stream reset" {
-			_ = s.Close()
-			return
-		} else if err != nil {
-			logger.Errorf("failed to decode list of peers err=%s", err)
-			continue
-		}
+	r := bytes.NewReader(buf)
+	peers := []peer.AddrInfo{}
+	dec := json.NewDecoder(r)
+	err = dec.Decode(&peers)
+	if err != nil {
+		logger.Errorf("failed to decode list of peers err=%s", err)
+		_ = s.Close()
+		return
+	}
 
-		for _, peer := range peers {
-			go func() {
-				connInfo := h.ConnectionManager.GetInfo()
-				if connInfo.ConnCount < connInfo.LowWater {
-					err = h.Host.Connect(h.Ctx, peer)
-					if err != nil {
-						logger.Errorf("failed to connect to peer err=%s", err)
-					} else {
-						logger.Debugf("connected to peer! %s", peer)
-					}
+	for _, peerp := range peers {
+		var peer peer.AddrInfo
+		peer = peerp
```

Comment:
you can also pass this into the go func, eg `go func(peer peer.AddrInfo)` then in the `()` that calls it, pass `peer`